### PR TITLE
Distributed signatures (EIP-11) implementation 

### LINKED ISSUE #1035: [BUG] Incorrect calculation of cost in ergo-wallet
### Problem
Ergo-wallet is computing transaction cost incorrectly. 
1) This https://github.com/ergoplatform/ergo/blob/8524b0ac85b27d480768f6e81e6102cf9d7dc096/ergo-wallet/src/main/scala/org/ergoplatform/wallet/interpreter/ErgoProvingInterpreter.scala#L55
should be moved outside the loop
2) current accumulated cost should be passed as initCost argument of Context

### PR DESCRIPTION
**Pull request by kushti**
Also close #1035

### CODE REVIEW

**Review by knizhnik — APPROVED** [review]

File: src/main/resources/api/openapi.yaml [hunk 1]
```diff
@@ -864,6 +867,175 @@ components:
...
+          description:  Hex-encoded big-endian 256-bits secret exponent
+          type: string
+          example: '433080ff80d0d52d7f8bfffff47f00807f44f680000949b800007f7f7ff1017f'
+        a:
+          description: Hex-encoded "g" generator for the Diffie-Hellman tuple (secp256k1 curve point)
+          type: string
+          example: '02a7955281885bf0f0ca4a48678848cad8dc5b328ce8bc1d4481d041c98e891ff3'
+
+    HintExtractionRequest:
+      description: request to extract prover hintsfrom a transaction
```
Comment (greenhat) [inline src/main/resources/api/openapi.yaml:887]:
```suggestion
      description: request to extract prover hints from a transaction
```

**Review by greenhat — APPROVED** [review]
LGTM

**Review by aslesarenko — APPROVED** [review]

File: ergo-wallet/src/main/scala/org/ergoplatform/wallet/interpreter/TransactionHintsBag.scala [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package org.ergoplatform.wallet.interpreter
+
+import sigmastate.interpreter.{HintsBag, OwnCommitment}
+
+case class TransactionHintsBag(secretHints: Map[Int, HintsBag], publicHints: Map[Int, HintsBag]) {
+
+  def putHints(index: Int, hintsBag: HintsBag): TransactionHintsBag = {
```
[the remark this answers is not in the document]
Comment (kushti) [reply ergo-wallet/src/main/scala/org/ergoplatform/wallet/interpreter/TransactionHintsBag.scala:13]: done!

File: ergo-wallet/src/main/scala/org/ergoplatform/wallet/interpreter/TransactionHintsBag.scala [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    val (secret, public) = hintsBag.hints.partition(_.isInstanceOf[OwnCommitment])
+
+    TransactionHintsBag(secretHints.updated(index, HintsBag(secret)), publicHints.updated(index, HintsBag(public)))
+  }
+
+  def allHints(index: Int): HintsBag = {
```
[the remark this answers is not in the document]
Comment (kushti) [reply ergo-wallet/src/main/scala/org/ergoplatform/wallet/interpreter/TransactionHintsBag.scala]: done!

**Pull request merged**